### ISSUE TITLE
Decide in general thread/photo/comments UI hierarchy

### ISSUE THREAD

**Issue by asutula** [textileio/textile-mobile#505]
Just looking at our wish list for things related to photos, comments and likes... 

It seems like the only difference between the "card" for an image in the thread view and the card for an image in the single image view is that the single image view will display all comments instead of just the caption and possibly a recent comment or two. I'm basing this on our general seeming desire to be able to do all the things from all the places.

Here's a proposal to provide more hierarchy within this part of the UX:

Image card in thread view:
- displays photo
- displays caption
- displays like count with option to like the photo
- (possibly) displays a recent comment
- (possibly) allow for comment creation
- displays comment count like "View all 23 comments" which opens the Comments screen:

Comments screen:
- displays a small thumb of the photo, and caption
- displays all comments
- allows comment creation

(Eventual) Image card in single image screen
- Could be used for display after opening a notification
- Basically the same as the thread feed card, but displays more recent comments
- We can use the card from the thread feed to get started

Main thing here is getting rid of the single image view as something you can open from the thread feed.

**Comment by carsonfarmer** [comment]
This looks good to me aaron, only thing i'd 'change' is i'm not sure we'd really need "(possibly) displays a recent comment" on the image card in a thread few... the comment count and "(possibly) allow for comment creation" would probably be sufficient?

**Comment by asutula** [comment]
I agree, just saw that @sanderpick had asked for a couple recent comments there, so just didn't want to rule it out without a little discussion.

**Comment by sanderpick** [comment]
Makes sense to me @asutula. I do think though we should aim to show a comment or two in the feed view (eventually, though given the structure here, might not be much more work). My thinking there is that it will increase engagement when you're just perusing the feed. It's not really natural to click into every single photo to see what the convo is... this way you can sorta "eves-drop".

**Comment by sanderpick** [comment]
This one doesn't seem necessary: 

- (possibly) allow for comment creation

**Comment by sanderpick** [comment]
For this one:

- displays comment count like "View all 23 comments" which opens the Comments screen

What if there are on comments? There should still be a way to get to the comments screen. IG solves this by just having a comment bubble next to the heart which opens the comments screen.

**Comment by asutula** [comment]
I think that Instagram's strategy here (my ideas are mostly from IG) is writing the comment is the engagement you want to create in the thread feed or in any/every screen because it's the action that keeps the photo poster feeling engaged and wanting to post again. Reading other people's comments is interesting, but secondary.

**Comment by asutula** [comment]
Your point about there being no comments... that's another reason it's nice to have comment creation in the thread feed card; No need for a comments screen if there are no comments.

**Comment by sanderpick** [comment]
Ok, it just seems like an obvious nice-to-have: "what's the latest comment here?" without having to go to a new screen. We can defer for now

**Comment by sanderpick** [comment]
Right yeah, I'm going off IG too... may as well just do what they have done until we find it not working.

**Comment by sanderpick** [comment]
So, you're saying a user should be able to comment from the thread card, but not see their comment they just wrote from thread card? Sounds tough to me.

**Comment by asutula** [comment]
No, not saying that. Tries to imply some amount of flexibility with my (possibly) label... You comment on the photo in the thread feed... This would be one case where it would make sense to display a comment there.

**Comment by asutula** [comment]
I think IG tends to display more bits of information in the thread view if/when users you follow liked or commented on a photo. For example, show up to two recent comments by people I follow, or show the usernames of two recent likes by people I follow. I'm not sure how that translates to our system since everyone participating in a thread "follows" everyone else in the thread implicitly. I guess it just comes down to recent comments/likes then.

**Comment by asutula** [comment]
I guess the closest thing we have to the idea of "following" someone comes down to who invited who to participate in a thread. But I don't know if we track that?

**Comment by asutula** [comment]
I think in our system, it's safe to say that everyone follows everyone and that we can simply display N recent comments/like usernames in the thread image card. I'm down with that.

**Comment by sanderpick** [comment]
I guess it would just be by time (recent). Okay, well... I trust your judgement. We can always iterate.

**Comment by sanderpick** [comment]
Okay, cool. Makes sense to me. Down to chat through it as well.

**Comment by asutula** [comment]
Ok sounds like we have a plan here. Going to ticket it up and delete some of the old UI nit notes that no longer apply. Speak now or forever hold your peace.

**Comment by asutula** [comment]
Decisions in here have been tracked in issues:

#511
#512
#509
#508
#497 
#507
#510
